fendermint/testing/materializer/src/docker/dropper.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use bollard::{
+    container::{RemoveContainerOptions, StopContainerOptions},
+    Docker,
+};
+
+/// Timemout before we kill the container if it doesn't want to stop.
+const KILL_TIMEOUT_SECS: i64 = 5;
+
+/// Commands to destroy docker constructs when they go out of scope.
+pub enum DropCommand {
+    DropNetwork(String),
+    DropContainer(String),
+}
+
+pub type DropHandle = tokio::sync::mpsc::UnboundedSender<DropCommand>;
+
+/// Start a background task to remove docker constructs.
+///
+/// The loop will exit when all clones of the sender channel have been dropped.
+pub fn start(docker: Docker) -> DropHandle {
+    let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
```

Comment:
Ah, so this is a nice Rust way to implement the producer/consumer pattern between async tasks, nice!

Reply:
Yes, I'd say this is the more common practice, and then with the `select!` macro in `tokio` you can do a whole lot of stuff like prioritising certain queues, doing timeouts, like [here](https://github.com/consensus-shipyard/ipc/blob/b2bdb1f7fd192e7d678cb19bacb92d2b17e1d848/ipld/resolver/src/service.rs#L267). The STM stuff is way less common. What I did with the runtime handle compiled but didn't run.